flang/lib/Lower/Bridge.cpp

```diff
@@ -1922,6 +1951,20 @@ class FirConverter : public Fortran::lower::AbstractConverter {
 
     // Reinstate entry block as the current insertion point.
     builder->setInsertionPointToEnd(&func.front());
+
+    if (hasAlternateReturns) {
+      // Create a local temp to hold the alternate return index.
+      // Give it an integer index type and the subroutine name (for dumps).
+      // Attach it to the subroutine symbol in the localSymbols map.
+      // Initialize it to zero, the "fallthrough" alternate return value.
+      const auto &symbol = funit.getSubprogramSymbol();
+      const auto altResult = builder->createTemporary(
+          toLocation(), builder->getIndexType(), symbol.name().ToString());
+      addSymbol(symbol, altResult);
```

Comment:
The functionality is similar, but I don't believe existing instances provide the needed functionality.  It might be possible to instead create a new 1-off function instance.

Reply:
`createTemp` handles the common case, but it doesn't work here, because it assumes that the `symbol` has a type that can be used.  There are other cases where `createTemporary` must be used directly instead.  This is one of them.